src/hdmf/build/map.py

```diff
@@ -1353,7 +1353,14 @@ def __get_type(self, spec):
         if spec.data_type_def is not None:
             return spec.data_type_def
         if spec.data_type_inc is not None:
-            return spec.data_type_inc
+            container_type = None
+            for val in self.__container_types.values():
+                container_type = val.get(spec.data_type_inc)
+                if container_type is not None:
+                    return container_type
+            if container_type is None:
+                # it is not clear when this would ever happen
```

Comment:
let's throw a warning here, since the container will not be set as a child. I think if this case ever comes up we can ask the user to define a custom class, rather than fail in a very subtle way.

Reply:
actually when I try to change this the code no longer works. My guess is the first time around the class you are trying to fetch may not be created yet so a string is used as a place-holder. The second time around all classes have been made, so it can properly place the class in docval instead of the string. Does this sound right?